python_ta/reporters/node_printers.py

```diff
@@ -389,23 +392,26 @@ def render_pep8_errors_e275(msg, line, col, source_line=None):
     )
 
 
-def render_pep8_errors_e301(msg, line, col, source_line=None):
+def render_pep8_errors_e301(msg, source_lines=None):
     """Render a PEP8 expected 1 blank line message."""
-    line -= 1
-    body = source_line[line]
+    line = msg.line - 1
+    yield from render_context(line - 1, line + 1, source_lines)
+    body = source_lines[line]
     indentation = len(body) - len(body.lstrip())
     yield (
         None,
         slice(None, None),
         LineType.ERROR,
         body[:indentation] + NEW_BLANK_LINE_MESSAGE,
     )
+    yield from render_context(msg.line, msg.line + 2, source_lines)
 
 
-def render_pep8_errors_e302(msg, line, col, source_line=None):
+def render_pep8_errors_e302(msg, source_lines=None):
     """Render a PEP8 expected 2 blank lines message."""
-    line -= 1
+    line = msg.line - 1
     if "found 0" in msg.msg:
+        yield from render_context(line - 1, line + 1, source_lines)
```

Comment:
Overall, the same number of lines of context (3) should be shown for all error messages. This means you'll need to update the custom renderers that use `source_lines` directly, so that they're consistent with generic approach (Lines 158-160).

In general, this code can be simplified to eliminate some +1/-1 adjustments and make the math more straightforward.

Set `line = msg.line` (the standard path on Line 156). Write all `render_context` calls in terms of `line` (not `msg.line`, like on Line 429). Eliminate the `line -= 1` statement on Line 425 (changing variable values makes code more complex).